### PR TITLE
Upgraded typehints

### PR DESCRIPTION
**Pull request by GameDungeon**
## Overview: What does this pull request change?
First, all code in the pull request was written by automated processes.
I believe that importing from animations will have great benefits for our typing efforts. 

An isort command was called to add the imports.
Black and pyupgrade took it from there.

The line "from \_\_future__ import annotations" was added to every python file to allow for 3.11 style annotations.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by Darylgolden — APPROVED** [review]
I like the new syntax, and if everything is done via `pyupgrade` I don't think there is really a risk of breaking anything. For the sake of the changelog and other reviewers, can you be more explicit on what exactly the changes made here were?

**Comment by GameDungeon** [comment]
This better?

**Comment by hydrobeam** [comment]
What are the specific features that this adds? What changes does this make to how types are written? What does `from __future__ import annotations` actually do?  Could you provide more explanation about the motivation, consequences etc... 

It feels like there's a lot of information that we're missing here.

**Comment by GameDungeon** [comment]
The real issue with that is I would have to rewrite several peps to explain everything.

**Comment by hydrobeam** [comment]
I don't think explaining the changes in your PR and why you made them amounts to "rewriting several peps".

**Comment by GameDungeon** [comment]
Well, the changes I made are. I told isort to add "from __future__ import annotations" to the top of every python file. Then pyupgrade upgraded them to the new sytanx.

**Comment by GameDungeon** [comment]
It seems to me the only thing holding this back is the change description. I honestly don't know what needs to be put, so I'll leave that to anyone else that wants a shot at it.

File: example_scenes/advanced_tex_fonts.py
Comment (Darylgolden) [inline example_scenes/advanced_tex_fonts.py:1]: I don't think example scenes should have the import. Please remove them from all example scenes.

**Review by Darylgolden — CHANGES_REQUESTED** [review]

File: example_scenes/basic.py
Comment (GameDungeon) [inline example_scenes/basic.py:3]:
```suggestion
```

File: example_scenes/customtex.py
Comment (GameDungeon) [inline example_scenes/customtex.py:1]:
```suggestion
```

File: example_scenes/opengl.py
Comment (GameDungeon) [inline example_scenes/opengl.py:1]:
```suggestion
```

**Review by GameDungeon — COMMENTED** [review]
Fixed the example scenes.

**Review by Darylgolden — APPROVED** [review]
I think this LGTM, the unnecessary imports can be removed in a separate PR.

**Pull request merged**
